From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Wed, 19 May 2021 00:10:29 +0200
Subject: tests: Do not create a file icon when no filename is provided

This would fail otherwise as GFileIcon requires a GFile parameter on
construction.

Origin: https://gitlab.gnome.org/GNOME/glade/-/merge_requests/114
---
 tests/create-widgets.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/create-widgets.c b/tests/create-widgets.c
index 2094059..db8311f 100644
--- a/tests/create-widgets.c
+++ b/tests/create-widgets.c
@@ -106,6 +106,8 @@ main (int   argc,
       if (G_TYPE_IS_INSTANTIATABLE (adaptor_type) && !G_TYPE_IS_ABSTRACT (adaptor_type) &&
           /* FIXME: can not create a themed icon without a name */
           !g_type_is_a (adaptor_type, G_TYPE_THEMED_ICON) &&
+          /* FIXME: can not create a file icon without a file name */
+          !g_type_is_a (adaptor_type, G_TYPE_FILE_ICON) &&
           /* FIXME: GtkPopoverMenu gives a few warnings */
           !g_type_is_a (adaptor_type, GTK_TYPE_POPOVER_MENU) &&
           /* FIXME: GtkFileChooserNative is hard to test here */
